pkg/apis/httpEndpoint/v1alpha1/types.go

```diff
@@ -51,11 +61,61 @@ func (h HTTPEndpoint) NameValuePairs() []common.NameValuePair {
 	return h.Spec.Headers
 }
 
+// nil returns a bool indicating if a client key has a secret reference
+func (h HTTPEndpoint) HasTLSClientKeySecret() bool {
+	return h.Spec.ClientKey != nil && h.Spec.ClientKey.SecretKeyRef != nil && h.Spec.ClientKey.SecretKeyRef.Name != ""
+}
+
+// HasTLSClientCertSecret returns a bool indicating if a client cert has a secret reference
+func (h HTTPEndpoint) HasTLSClientCertSecret() bool {
+	return h.Spec.ClientCert != nil && h.Spec.ClientCert.SecretKeyRef != nil && h.Spec.ClientCert.SecretKeyRef.Name != ""
+}
+
+// HasTLSRootCASecret returns a bool indicating if a root cert has a secret reference
+func (h HTTPEndpoint) HasTLSRootCASecret() bool {
+	return h.Spec.RootCA != nil && h.Spec.RootCA.SecretKeyRef != nil && h.Spec.RootCA.SecretKeyRef.Name != ""
+}
+
+// HasTLSRootCA returns a bool indicating if the HTTP endpoint contains a root ca
+func (h HTTPEndpoint) HasTLSRootCA() bool {
+	return h.Spec.RootCA != nil && h.Spec.RootCA.Value != nil
+}
+
+// HasTLSClientCert returns a bool indicating if the HTTP endpoint contains a client cert
+func (h HTTPEndpoint) HasTLSClientCert() bool {
+	return h.Spec.ClientCert != nil && h.Spec.ClientCert.Value != nil
+}
+
+// HasTLSClientKey returns a bool indicating if the HTTP endpoint contains a client key
+func (h HTTPEndpoint) HasTLSClientKey() bool {
+	return h.Spec.ClientKey != nil && h.Spec.ClientKey.Value != nil
+}
```

Comment:
Same here re: returning `true`, even if the `Value` is nil.

Reply:
Commented above